src/components/DiagramFrame/SeqDiagram/MessageLayer/MessageLabel.vue

```diff
@@ -0,0 +1,64 @@
+<template>
+  <label
+    title="Double click to edit"
+    class="px-1 cursor-text right hover:text-skin-message-hover hover:bg-skin-message-hover"
+    :class="{
+      'absolute right-1/2 translate-x-1/2 bottom-0  py-1 px-2 ml-1 cursor-text':
```

Comment:
`absolute right-1/2 translate-x-1/2`  rules ensure that the editing text input is centered and does not push the message arrow out of position.